subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/RealLifeAndroidDexingTransformsPerformanceTest.groovy

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.regression.android
+
+import org.gradle.performance.AbstractCrossBuildPerformanceTest
+import org.gradle.performance.fixture.BuildExperimentInvocationInfo
+import org.gradle.performance.fixture.BuildExperimentListenerAdapter
+import org.gradle.performance.fixture.GradleInvocationSpec
+import org.gradle.performance.results.BaselineVersion
+import org.gradle.performance.results.CrossBuildPerformanceResults
+import org.gradle.util.GFileUtils
+import spock.lang.Unroll
+
+class RealLifeAndroidDexingTransformsPerformanceTest extends AbstractCrossBuildPerformanceTest {
+
+    private static final String DEXING_TRANSFORM = "dexing transform"
+    public static final String DEXING_TASK = "dexing task"
+
+    @Unroll
+    def "dexing task vs transform: #tasks on #testProject without (android/transform) cache"() {
+        given:
+        def invocationOptions = [tasks: tasks, memory: memory, enableAndroidBuildCache: false]
+
+        runner.testGroup = "Android dexing"
+        runner.buildSpec {
+            projectName(testProject)
+            displayName(DEXING_TRANSFORM)
+            warmUpCount warmUpRuns
+            invocationCount runs
+            listener(cleanTransformsCache())
+            invocation {
+                defaultInvocation(*:invocationOptions, dexingTransforms: true, delegate)
+            }
+        }
+
+        runner.baseline {
+            projectName(testProject)
+            displayName(DEXING_TASK)
+            warmUpCount warmUpRuns
+            invocationCount runs
+            listener(cleanTransformsCache())
+            invocation {
+                defaultInvocation(*:invocationOptions, dexingTransforms: false, delegate)
+            }
+        }
+
+        when:
+        def results = runner.run()
+
+        then:
+        results.assertEveryBuildSucceeds()
+        and:
+        assertDexingTransformIsFaster(results)
+
+        where:
+        testProject         | memory | warmUpRuns | runs | tasks
+        'largeAndroidBuild' | '5g'   | 2          | 4    | 'clean assembleDebug'
+    }
+
+    @Unroll
+    def "dexing task vs transform: #tasks on #testProject"() {
```

Comment:
The only real difference between the two tasks is `enableAndroidBuildCache` being set to `true` vs `false` and whether or not we register the cache cleanup listener. I think we should merge them.

Reply:
I merged the two tests.